src/internal.cc

```diff
@@ -86,8 +86,7 @@ internalStaticCheck(const SBuf &urlPath)
 bool
 ForSomeCacheManager(const SBuf &urlPath)
 {
-    static const SBuf mgrPfx("/squid-internal-mgr");
-    return urlPath.startsWith(mgrPfx);
+    return urlPath.startsWith(CacheManager::MagicUrlPathPrefix());
 }
```

Comment:
> PR description: ... Denying them properly results in simpler/smaller code (than allowing them would), so we should avoid a UI change and continue to deny them, at least for now.

We have actually implemented code that would start allowing "slashless" cache manager URLs, so the above claim is not a speculation, but avoiding the UI change is the primary reason to continue to deny them in this small bug-fixing PR.

Reply:
Not sure I follow what you are saying here. IF you already have a proper fix implemented why bother with a hack.

FWIW ...

1) `/squid-internal-mgr` without a trailing slash semantically means that the request reaches the mgr, but the mgr logic does not have any resource defined for that URI. So it should result in 404.

2) this PR is going into `master` branch. Avoiding UI change is not a consideration for code going there - just add appropriate documentation in the PR.

Reply:
>> We have actually implemented code that would start allowing "slashless" cache manager URLs

> IF you already have a proper fix implemented why bother with a hack

This PR does not contain a hack. It contains a proper bug fix for a recently introduced bug.

Properly handling slashless URLs (after they are allowed) requires a lot more than just "allowing" slashless URLs. However, just allowing that URL alone already requires quite a bit of code changes. We know the latter because we have implemented those changes. The changes we implemented are not enough for a "proper fix", but any "proper fix" will contain similar changes (and a lot more).

>> avoiding the UI change is the primary reason

> Avoiding UI change is not a consideration for code going [into `master` branch].

It is a consideration for PR scope. This is a minimal PR fixing a very recent bug by restoring very old pre-bug behavior. Nothing else. We could do a lot more, but we do not have to. The current minimal scope is reasonable.